### PR TITLE
unique uses FxHash instead of SipHash

### PR DESCRIPTION
**Pull request by zzlk**
micro/ops/unique        time:   [84.333 µs 84.933 µs 85.651 µs]
                        change: [-68.397% -68.069% -67.798%] (p = 0.00 < 0.05)
                        Performance has improved.

### CODE REVIEW

File: hydroflow/src/lang/clear.rs [hunk 1]
```diff
@@ -46,7 +47,9 @@ clear_impl!(BTreeMap<K, V>);
 clear_impl!(BTreeSet<T>);
 clear_impl!(BinaryHeap<T>);
 clear_impl!(HashMap<K, V>);
+clear_impl!(FxHashMap<K, V>);
 clear_impl!(HashSet<T>);
+clear_impl!(FxHashSet<T>);
```
Comment (MingweiSamuel) [inline hydroflow/src/lang/clear.rs:52]: Would it work to just change the existing impl to add a hasher?
```rust
clear_impl!(HashMap<K, V, S>);
clear_impl!(HashSet<T, S>);
```

File: hydroflow/src/lang/monotonic_map.rs [hunk 2]
```diff
@@ -27,6 +30,7 @@ where
 impl<K, V> MonotonicMap<K, V>
 where
     K: PartialOrd,
+    V: Clear,
```
Comment (MingweiSamuel) [inline hydroflow/src/lang/monotonic_map.rs:33]: Would be a good optimization to use clear but it looks like you aren't actually doing anything with it right now?

**Review by MingweiSamuel — CHANGES_REQUESTED** [review]
Looks like there's some loose `Clear` code that isn't used but might be good to actually use

File: hydroflow/src/lang/monotonic_map.rs (hunk 2 above)
Comment (zzlk) [reply hydroflow/src/lang/monotonic_map.rs:33]: This change is just restoring the code to what it was before my change, I didn't know about clear being our own trait that could be impld when I removed the trait bounds.

Comment (MingweiSamuel) [reply hydroflow/src/lang/monotonic_map.rs:33]: Hmm ok

File: hydroflow/src/lang/clear.rs (hunk 1 above)
Comment (zzlk) [reply hydroflow/src/lang/clear.rs:52]: ya it does work

**Comment by zzlk** [comment]
> Looks like there's some loose `Clear` code that isn't used but might be good to actually use

I'll make a github issue so we can follow up on it more comprehensively.

**Review by shadaj — APPROVED** [review]

**Pull request merged**